fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java

```diff
@@ -673,11 +673,19 @@ public List<HdfsFileDesc> getHdfsFileDescs(String dirPath, boolean isSplittable,
         // block locations of the files in the given path in one operation.
         // The performance is better than getting status and block location one by one.
         try {
-            RemoteIterator<LocatedFileStatus> blockIterator = fileSystem.listLocatedStatus(new Path(uri.getPath()));
+            // files in hdfs may have multiple directories, so we need to list all files in hdfs recursively here
+            RemoteIterator<LocatedFileStatus> blockIterator = null;
+            if (!Config.recursive_dir_search_enabled) {
+                blockIterator = fileSystem.listLocatedStatus(new Path(uri.getPath()));
+            } else {
+                blockIterator = fileSystem.listFiles(new Path(uri.getPath()), true);
+            }
             while (blockIterator.hasNext()) {
                 LocatedFileStatus locatedFileStatus = blockIterator.next();
                 if (!isValidDataFile(locatedFileStatus)) {
                     continue;
+                } else if (locatedFileStatus.isDirectory() && Config.recursive_dir_search_enabled) {
```

Comment:
I have checked the listFiles(true)'s behavior. It will return the files and the subDir's files. So I think there is no need to check the isDir?

Reply:
> I have checked the listFiles(true)'s behavior. It will return the files and the subDir's files. So I think there is no need to check the isDir?

As this pr's target, fe should scan hdfs files recursively only when config set `recursive_dir_search_enabled` to true.So,if locatedFileStatus is a dir, and fe enable `recursive_dir_search_enabled`, we check dir here. If recursive_dir_search_enabled is false, fe shouldn't scan sub dir of hdfs.